### ISSUE TITLE
Feature: Give the option to use Neovim as a backend

### ISSUE THREAD

**Issue by Tylerc230** [XVimProject/XVim#1043]
Neovim is able to be used as a headless backend without a UI. This means that it may be possible to route keypresses from xcode to neovim and show neovim's buffer in xcode. This would allow access to all vim plugins and built in features in Xcode.

**Comment by JugglerShu** [comment]
Yea that kind of thing is considered. Actually it is possible to pass all the buffer to Neovim or Vim and execute command and apply back the result to Xcode, but I think this is not really smart solution since this would be disable to use Xcode features like auto complete. That is the main reason I'm not doing it.

**Comment by Tylerc230** [comment]
Yeah makes sense. I guess the tradeoffs are using vim/neovim plugins in
xcode but not xcode auto complete features.

**Comment by Tylerc230** [comment]
Some of Xcode's completion engine is open source. It's a matter of creating
the vim plugins for it.

**Comment by fiveNinePlusR** [comment]
@Tylerc230 do you know where that is documented?

**Comment by Tylerc230** [comment]
https://github.com/apple/swift/tree/master/tools/SourceKit/docs <https://github.com/apple/swift/tree/master/tools/SourceKit/docs>

**Comment by fiveNinePlusR** [comment]
ah yeah. I have used sourcekitten for completion before for completion in emacs. 

https://github.com/jpsim/SourceKitten

**Comment by Tylerc230** [comment]
Yep built in source kit. I would love to collaborate with some folks on
building some vim plugins for swift.

**Comment by fiveNinePlusR** [comment]
there is this https://github.com/mitsuse/autocomplete-swift

**Comment by Tylerc230** [comment]
Yeah there's also deoplete-swift. I don't think either supports multiple
files/Xcode workspaces
